pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -150,7 +154,12 @@ func (r *Reconciler) onError(
 	objs []*unstructured.Unstructured,
 ) (reconcile.Result, error) {
 	// settting overall status to failed
-	r.setStatus(sbrStatus, bindingFail)
+	conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+		Type: conditions.BindingFailed
+		Status: conditions.ConditionFalse
+		Reason: "Binding Failed"
```

Comment:
`Reason` is part of the API, and it should be encoded in PascalCase. In this particular case, it should unroll the error to use a more specific reason, and `Message` can contain the error message.